libraries/chain/include/eosio/chain/wasm_interface_private.hpp

```diff
@@ -380,7 +385,9 @@ struct intrinsic_invoker_impl<Ret, std::tuple<array_ptr<T>, size_t, Inputs...>,
    static Ret translate_one(wasm_interface &wasm, Inputs... rest, Translated... translated, I32 ptr, I32 size) {
       auto mem = getDefaultMemory(intrinsics_accessor::get_context(wasm).code.instance);
       size_t length = size_t(size);
-      T *base = memoryArrayPtr<T>(mem, ptr, length);
```

Comment:
How come we don't keep the use of memoryArrayPtr and pull these checks into it?

Reply:
eh my thought was to not the change the library unless needed. In this case there is already an API there to do what is desired, so figured to use what is already there.